File: backend/app/models/receipt.py
import uuid

from sqlalchemy import ForeignKey, String
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db import Base
from app.models.base import TimestampMixin, UUIDMixin


class Receipt(UUIDMixin, TimestampMixin, Base):
    __tablename__ = "receipts"

    expense_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("expenses.id", ondelete="CASCADE"), nullable=False
    )
    # MinIO object reference; the app fetches bytes via the API (GET /receipts/{id}/content)
    bucket: Mapped[str] = mapped_column(String(128), nullable=False)
    object_key: Mapped[str] = mapped_column(String(512), nullable=False)
    content_type: Mapped[str | None] = mapped_column(String(128), nullable=True)

    expense: Mapped["Expense"] = relationship(back_populates="receipts")  # noqa: F821

File: ios/SplitBackAPI/Sources/SplitBackAPI/Models/Receipt.swift
import Foundation
import SwiftData

/// A receipt image stored in MinIO and registered against an expense. Mirrors `receipts`.
/// The app references objects by `bucket`/`objectKey` and fetches bytes via presigned URLs.
@Model
final class Receipt {
    @Attribute(.unique) var id: UUID
    var bucket: String
    var objectKey: String
    var contentType: String?
    var createdAt: Date
    var expense: Expense?

    init(
        id: UUID,
        bucket: String,
        objectKey: String,
        contentType: String? = nil,
        createdAt: Date,
        expense: Expense? = nil
    ) {
        self.id = id
        self.bucket = bucket
        self.objectKey = objectKey
        self.contentType = contentType
        self.createdAt = createdAt
        self.expense = expense
    }
}
